### ISSUE TITLE
Instructions are unclear for less-experienced users

### ISSUE THREAD

**Issue by e-t-l** [etesync/server-skeleton#21]
I'm fairly computer-savvy but still learning, so I'm currently unfamiliar with some of the processes described in the readme. In other words, I tried to set up a self-hosted etesync server, it didn't work, and I have no idea why because the instructions are so bare-bones. Here's what I did (trying to install on Windows 10)...

> git clone https://github.com/etesync/server-skeleton.git
> cd server-skeleton
> #Set up the environment and deps
> virtualenv -p python3 venv  # If doesn't work, try: virtualenv3 venv
> source venv/bin/activate
> pip install -r requirements.txt

So far so good. Instructions said Windows users could skip installing virtualenv, so I went straight from `cd server-skeleton` to `pip install -r requirements.txt`. I also went ahead and renamed the easy config file because I'm not familiar with Django. 

> manage.py migrate

This output a nice list of "Applying [whatever]... OK" with no errors so I'm assuming that did what it's supposed to.

> manage.py runserver 0.0.0.0:8000

This outputs the following:
`System check identified no issues (0 silenced).`
`[this is just the date and time]`
`Django version 2.1.12, using settings 'etesync_server.settings'`
`Starting development server at http://0.0.0.0:8000/`
`Quit the server with CTRL-BREAK`
And honestly I have no idea what to do from here. Opening "http://0.0.0.0:8000/" in a web browser just throws an "unable to connect" page. The readme says to run the debug server to make sure everything works as expected, but it's unclear what that means, or what to do if it doesn't.

The links in the Production Deployment section were not particularly helpful, as the first requires an in-depth understanding of Django, and while the second looks more user-friendly, there's a major roadblock: It is a [known issue ](https://stackoverflow.com/questions/40268566/uwsgi-fails-to-install-on-windows)that uwsgi can't be installed on Windows. If we move on from here:

> There are more details about a proper production setup in the wiki.

No, there aren't. 

And lastly:

> manage.py createsuperuser

Thankfully, this step actually appeared to work. However, it's functionally useless if the server remains inaccessible. 

Etesync's documentation assumes everyone has the same level of experience and knowledge as the developer. I'll admit that I might be less knowledgeable than the average etesync user, but this is an unreasonably high barrier to entry. The User Guide for the app is much more informative and accessible, and it would be nice if the instructions here were at a reasonably similar level. 

I hope this doesn't come across as too rant-y, because I know the devs are hardworking people. I don't expect you to write out every step for the person who has never touched a computer before. But a bit more explanation and description would go a long way. At the very least, it might be a good idea to list of the programs that will be needed in order to accomplish this. The documentation indicates that this can be set up on any system that can run Python, and while that's true, it doesn't mention that you'll also need to be familiar with Django, nginx, uWBSI or WBSI or possibly Apache, etc. At least users would know whether they have the skills and software required to do it successfully. 

Ideally, there would be a pre-built installation package for Windows, but again I understand that the devs are only human, and building packages takes time. Thanks for taking the time to read my pseudo-rant, and I hope it has provided some useful and not-too-annoying feedback.

**Comment by tasn** [comment]
You did everything right, the only thing missing is that "0.0.0.0:8000" means that you can access the server on any address that leads to this computer. So from the same computer you can use: localhost:8000, from another computer, you'd need to use your computer's IP address.

I understand where you are coming from with the instructions being raw and require a lot of prior knowledge - because it's true. There is A LOT to cover when it comes to setting up your own server, and as you briefly mentioned, there are also a lot of alternatives. For example nginx vs Apache. Trying to cover all of them is just infeasible, and anyhow, even if we do manage, there are a lot of implications for every decision you make. The surface area is just too large.
Another such example: even if you got everything working, you'd probably need to set up port forwarding so your server can be accessed from the outside internet. Same with opening firewall ports. There really is just too much, and your setup (of having the *server* run on Windows) is rather unique which doesn't help. :)

As for packages: we wish, though it's a lot of work as you said and for not a lot of benefit. Only recently we added packages for some popular Linux distros and even that was a community contribution.

With that being said, if you see an easy way to improve it (now that you got it working), please open a PR! We would love to improve!

I'm closing this as I assume it works (the only thing you have missing is knowing which IP to connect to). Please reopen if it doesn't.

**Comment by e-t-l** [comment]
Thanks for the fast response. I should have mentioned, in addition to trying 0.0.0.0:8000 in the browser, I did also already try localhost:8000 and [my local IP]:8000, but both of those returned "Bad Request (400)" errors (that was why I thought I was missing a step in regards to nginx, port-forwarding, etc). However, when I navigated to localhost:8000, I did notice that my console output the following:
`[date & time] "GET / HTTP/1.1" 400 26`

And navigating to [my local IP]:8000 output:
`[date & time] "GET / HTTP/1.1" 400 26`
`[date & time] "GET /favicon.ico HTTP/1.1" 400 26`

Unfortunately, I don't know what that means. I'm guessing I'm successfully connecting to the port, but there's something wrong with my install? Not sure how to troubleshoot further though.

(Btw, I totally understand why making a Win package isn't worth the effort. As you can tell, I'm not the most experienced programmer, but I'm trying to run etesync as a pet project to learn from the process. Ideally, I would have a dedicated Linux device to host on, but since a Windows desktop is all I have at the moment, that's what I'm trying to make work.)

**Comment by e-t-l** [comment]
Update: After doing a bit of research into the 400 error I found this article: https://stackoverflow.com/questions/19875789/django-gives-bad-request-400-when-debug-false and realized I had configured allowed_host1 wrong in etesync-server.ini. Now when I navigate to localhost:8000 I get a message reading: "It works!" and instructions to read the readme. I assume that once I forward the port in my router and assign a URL to it, I should be able to plug that into the etesync app and connect.
